File: backends/ts-hono/src/domain/invitation/authRepository.ts
/**
 * 招待認証リポジトリインターフェースモジュール。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */

/** 招待認証トークンのキャッシュリポジトリインターフェース。 */
export interface InvitationAuthRepository {
  /** トークンとロールを指定秒数キャッシュします。 */
  store(token: string, role: number, ttl: number): Promise<void>;
  /** トークンに紐づくロールを取得します。存在しない場合は null を返します。 */
  find(token: string): Promise<number | null>;
  /** トークンを削除します。 */
  remove(token: string): Promise<void>;
}

File: backends/python-fastapi/app/domain/invitation/auth_repository.py
"""
招待認証リポジトリインターフェースモジュール。

Author: Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
"""
from abc import ABC, abstractmethod
from typing import Optional


class InvitationAuthRepository(ABC):
    """招待認証トークンのキャッシュリポジトリインターフェース。"""

    @abstractmethod
    def store(self, token: str, ttl: int) -> None:
        """トークンを指定秒数キャッシュします。"""

    @abstractmethod
    def find(self, token: str) -> Optional[str]:
        """トークンを取得します。存在しない場合は None を返します。"""

    @abstractmethod
    def remove(self, token: str) -> None:
        """トークンを削除します。"""
